### PR TITLE
AppItem: Prevent malformed URL to make the app crash

### PR DESCRIPTION
**Pull request by gregorylegarec**
~~When serving an app locally with option `--appdir`, the app is listed in the endpoint `/apps/` but all fields are `undefined`.~~

If a document has been created by mistake in DB `io.cozy.apps`, it is returned in the `/apps/` endpoint response with all its fields being `undefined`.

This PR ensure that an `undefined` href will not make the whole app to crash when building the app url.

### CODE REVIEW

**Comment by Crash--** [comment]
On peut avoir comment et pourquoi une URL mal formée ?

**Comment by edas** [comment]
> On peut avoir comment et pourquoi une URL mal formée ?

Ca me gêne aussi. Si on a un cas qui a cassé, il est plutôt urgent de l'identifier et le corriger en amont. Là on va ignorer des erreurs qui me semblent importantes.

**Comment by Crash--** [comment]
Woa, je n'avais pas vu que c'était marqué dans la description de la PR. Shame on me. 

Je pense que si on a une app qui est arrivée par mégarde dans la DB, alors c'est plutôt soit à la récup de la donnée qu'on doit faire attention à avoir toutes les données , soit savoir pourquoi elle est arrivée par mégarde ?

**Comment by gregorylegarec** [comment]
C'était en effet pour moi la prochaine étape, décider ou et quand assainir les données. Dès Cozy-Cient ? Dans l'app depuis un helper dans cozy-doctype ?

En attendant je pense que ça ne coûte pas trop cher de try/catch un appel qui peut de toutes façons throw une erreur, et éviter que l'app ne crashe. Ici le problème est survenu en dev, donc sans vraiment d'impact ni sur l'inté ou sur staging, mais ça m'a coûté un peu de temps et d'énergie de parvenir à comprendre la cause de ma page blanche.

File: src/components/Apps/AppItem.jsx [hunk 1]
```diff
@@ -30,7 +30,14 @@ export class AppItem extends React.Component {
...
 
   buildAppUrl(href) {
-    const url = new URL(href)
+    let url
+    try {
+      url = new URL(href)
+    } catch (error) {
+      // eslint-disable-next-line no-console
+      console.error(error.message)
+      return ''
```
Comment (ptbrowne) [inline src/components/Apps/AppItem.jsx]: `return null` to me is clearer. It will also prevent the AppLinker to add an empty href to the underlying element. Only a null prop which will not be trasnformed into an attribute.

Comment (gregorylegarec) [reply src/components/Apps/AppItem.jsx]: Ok. So we need more test to see how it behaves with an URL being null. 👍

**Comment by nono** [comment]
Ping @gregorylegarec

**Comment by nono** [comment]
Ping @gregorylegarec

File: src/components/Apps/AppItem.jsx (hunk 1 above)
Comment (gregorylegarec) [reply src/components/Apps/AppItem.jsx]: Done ! :tada:

**Comment by gregorylegarec** [comment]
Waiting for Travis to be green.

**Comment by ptbrowne** [comment]
It seems there is a problem with [bundle size](https://travis-ci.org/cozy/cozy-bar/builds/562203060#L1072) but everything is OK apart from that :)

:shipit:

**Comment by gregorylegarec** [comment]
@ptbrowne The problem with `bundle size` seems to have been solved with a rebase. Could you approve ? Thanks !
